extensions/presentation-tags-3.3.md

```diff
@@ -0,0 +1,103 @@
+clients. These tags (`intent`, `redirect`, `name`, and `log`) allow server
+and bot authors to specify that their messages carry particular meaning in
+a standardized way.
+
+## Motivation
+
+Over the years since RFC 1459, several meanings have been imputed to `NOTICE`
+beyond that specification's requirement that there be no automatic responses
+to messages sent via `NOTICE`. Specialized message formats have also become
+a de-facto standard for actions and for instructing clients to display
+messages in a particular context. 
+
+## Architecture
+
+### Capabilities
+
+This specification adds the `intents` capability, which enables all of the
+tags documented in this specification.
+
+### Tags
+
+#### The `intent` Tag
+
+The `intent` tag carries the sender's intentions on how the message should be
+interpreted. The following intentions SHOULD be supported:
+
+| Intention   | Meaning                                               |
+|-------------|-------------------------------------------------------|
+| `action`    | This message is an 'action'.                          |
+| `announce`  | Mark this message as an announcement.                 |
+| `info`      | This message is for informational purposes, e.g. news |
+| `error`     | This message is an 'error' notification.              |
+| `noreply`   | Do not generate a reply.                              |
+
+The `action` intent MAY be translated to and from the legacy CTCP encoding.
+
+#### The `redirect` tag
+
+The `redirect` tag specifies a target context which is different from the usual
+context. The format for the `redirect` tag is as follows:
+
+    redirect=<Target>
+
+The `<Target>` must be a valid channel name.
+
+#### The `log` Tag
+
+The `log` tag MAY carry a string indicating a severity. It is intended
+primarily for automatic reporting of server state. The format is as follows:
+
+    log=<String>
+
+#### The `name` Tag
+
+The `name` intent allows a user to represent a given message as being
+associated with a different name. In order to prevent abuse, implementations
+SHOULD provide mechanisms to prevent users from masquerading as server or
+channel operators, and clients MUST visually associate the user's actual
+handle with the masqueraded handle. It has the following format:
+
+    name=<String>
+
+### Examples
+
+`@intent=action PRIVMSG #example :dances a jig.`
+
+`@intent=announce,noreply PRIVMSG #example :Please be advised that this is a test.`
+
+`@intent=noreply,announce;redirect=#example PRIVMSG user :Follow the rules at...`
+
+`@log=info NOTICE user :Foobar opered up.
+
+### Legacy translation
+
+The `intent=action`, `intent=noreply`, and `redirect` tags SHOULD be
+translated as follows for legacy clients:
+
+* `intent=action` → `\x01ACTION ...`
+* `intent=noreply` becomes a `NOTICE`.
+* `redirect=<target>` → `NOTICE user :[<target>] ...`
+
+## Security Considerations
+
+There are no security considerations that implementations should note
```

Comment:
"In order to prevent abuse" from `name` sounds like a security consideration for me.
Also, `@redirect=#a nick!u@h PRIVMSG #b :foo` when used to message on channel even when banned there.